releasenotes/notes/fix-internal-writer-recreate-keep-extra-headers-dc8cd1d57c95d692.yaml

```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    internal: This fix resolves an issue where a writer recreate (for example in a subprocess) would not keep extra headers.
```

Comment:
I think we can drop the release note, this isn't _really_ a recommended way for customers to create/use an `AgentWriter`, so not sure we need to call it out. no strong opinion, but I wouldn't question it if there was no release note here.